### ISSUE TITLE
Fix 2.2.3 tests

### ISSUE THREAD

**Issue by TheKevJames** [TheKevJames/puppet-homebrew#48]
For some reason, the ruby 2.2.3 tests are failing and I can not for the life of me figure out why. Disabled 2.2.3 tests for now, since I've verified the module works locally with 2.2.3 and the tests seem to be travis-env related (maybe?), but it would be great if we could fix this and get them re-enabled.

[Travis build #176](https://travis-ci.org/TheKevJames/puppet-homebrew/builds/156624873) of commit c367a34654 is the last master build before these tests were disabled.

```
Debug: Listing installed packages
Debug: Executing '/usr/bin/stat -nf %Uu /usr/local/bin/brew'
Debug: Executing '/usr/bin/stat -nf %Ug /usr/local/bin/brew'
Debug: Executing '/usr/local/bin/brew list --versions'
Debug: Found packages /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:95:in `block in materialize': Could not find CFPropertyList-2.2.8 in any of the sources (Bundler::GemNotFound)
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `map!'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `materialize'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `map!'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `materialize'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
Warning: Could not match    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Warning: Could not match    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Error: Could not prefetch package provider 'brew': No resource and no name in property hash in brew instance
Debug: Listing installed packages
Debug: Executing '/usr/bin/stat -nf %Uu /usr/local/bin/brew'
Debug: Executing '/usr/bin/stat -nf %Ug /usr/local/bin/brew'
Debug: Executing '/usr/local/bin/brew list --versions git'
Debug: Found packages /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:95:in `block in materialize': Could not find CFPropertyList-2.2.8 in any of the sources (Bundler::GemNotFound)
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `map!'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `materialize'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `map!'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `materialize'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
Warning: Could not match    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Warning: Could not match    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Debug: Prefetching brewcask resources for package
Debug: Listing installed packages
Debug: Executing '/usr/bin/stat -nf %Uu /usr/local/bin/brew'
Debug: Executing '/usr/bin/stat -nf %Ug /usr/local/bin/brew'
Debug: Executing '/usr/local/bin/brew cask list --versions'
Debug: Found packages /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:95:in `block in materialize': Could not find CFPropertyList-2.2.8 in any of the sources (Bundler::GemNotFound)
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `map!'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `materialize'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `map!'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/spec_set.rb:88:in `materialize'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
Warning: Could not match    from /Users/travis/.rvm/rubies/ruby-2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
Warning: Could not match    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Warning: Could not match    from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
Error: Could not prefetch package provider 'brewcask': No resource and no name in property hash in brewcask instance
Debug: Listing installed packages
Debug: Executing '/usr/bin/stat -nf %Uu /usr/local/bin/brew'
Debug: Executing '/usr/bin/stat -nf %Ug /usr/local/bin/brew'
Debug: Executing '/usr/local/bin/brew cask list --versions'
Error: /Package[google-chrome]: Could not evaluate: invalid number of elements (14 for 1..2)
```

**Comment by jordigg** [comment]
I've seen similar error when brew doesn't get cloned.
```
Error: Could not prefetch package provider 'brew': No resource and no name in property hash in brew instance
Debug: Listing installed packages
Error: Could not prefetch package provider 'brewcask': No resource and no name in property hash in brewcask instance
Debug: Listing installed packages
```
Disappear when the brew command works on the terminal or you add `commands :brew => '/usr/local/bin/brew'` where required. Better to check that is getting cloned properly inside the path.

**Comment by TheKevJames** [comment]
This has been fixed in #71. Still figuring out why "slow tests" fail, but at least this issue is dealt with.
